src/rpdk/core/templates/docs-readme.md

```diff
@@ -0,0 +1,105 @@
+# {{ type_name }}
+{% if schema.description %}
+
+{{ schema.description }}
+{% endif %}
+
+## Syntax
+
+To declare this entity in your AWS CloudFormation template, use the following syntax:
+
+### JSON
+
+<pre>
+{
+    "Type" : "{{ type_name }}",
+    "Properties" : {
+{% if schema.properties %}
+{% for propname, prop in schema.properties.items() %}
+        "<a href="#{{ propname.lower() }}" title="{{ propname }}">{{ propname }}</a>" : <i>{{ prop.jsontype }}</i>{% if not loop.last %},{% endif %}
+
+{% endfor %}
+{% endif %}
+    }
+}
+</pre>
+
+### YAML
+
+<pre>
+Type: {{ type_name }}
+Properties:
+{% if schema.properties %}
+{% for propname, prop in schema.properties.items() %}
+    <a href="#{{ propname.lower() }}" title="{{ propname }}">{{ propname }}</a>: <i>{{ prop.yamltype }}</i>
+{% endfor %}
+{% endif %}
+</pre>
+{% if schema.properties %}
+
+## Properties
+
+{% for propname, prop in schema.properties.items() %}
+#### {{ propname }}
+{% if prop.description %}
+
+{{ prop.description }}
+{% endif %}
+
+_Required_: {% if propname in schema.required %}Yes{% else %}No{% endif %}
+
+
+_Type_: {{ prop.longformtype }}
+{% if prop.allowedvalues %}
+
+_Allowed Values_: {% for allowedvalue in prop.allowedvalues %}<code>{{ allowedvalue }}</code>{% if not loop.last %} | {% endif %}{% endfor %}
+
+{% endif %}
+{% if prop.minLength %}
+
+_Minimum_: <code>{{ prop.minLength }}</code>
+{% endif %}
+{% if prop.maxLength %}
+
+_Maximum_: <code>{{ prop.maxLength }}</code>
+{% endif %}
+{% if prop.pattern %}
+
+_Pattern_: <code>{{ prop.pattern }}</code>
+{% endif %}
+{% if prop.createonly %}
+
+_Update requires_: [Replacement](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks-update-behaviors.html#update-replacement)
+{% else %}
+
+_Update requires_: [No interruption](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-updating-stacks-update-behaviors.html#update-no-interrupt)
+{% endif %}
```

Comment:
If we're generating documentation, it's probably worth noting nuance for something as important as potential resource replacement. For example, a version property might be upgradable in-place but not downgradable in-place, so neither of these definitions would be entirely accurate. My assumption would be while schema definitions like `required` and `createOnlyProperties` guarantee a property is required or requires resource replacement to update respectively, other properties not specified as `required` or `createOnlyProperties` may in fact be required or cause resource replacement under some circumstances. [Immutability is important enough that its nuance should probably be captured in the resource schema itself in the future](https://github.com/aws-cloudformation/aws-cloudformation-resource-schema/issues/30)

Reply:
Real world example:
[`Updating AWS::EC2::VPC.InstanceTenancy requires no replacement only if you are updating its value from "dedicated" to "default". Updating AWS::EC2::VPC.InstanceTenancy from "default" to "dedicated" requires replacement`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-vpc.html#cfn-aws-ec2-vpc-instancetenancy)